google/cloud/aiplatform/metadata/experiment_resources.py

```diff
@@ -0,0 +1,717 @@
+#
+
+import abc
+from dataclasses import dataclass
+import logging
+from typing import Dict, List, NamedTuple, Optional, Union, Tuple, Type
+
+from google.auth import credentials as auth_credentials
+
+from google.cloud.aiplatform import base
+from google.cloud.aiplatform.metadata import artifact
+from google.cloud.aiplatform.metadata import constants
+from google.cloud.aiplatform.metadata import context
+from google.cloud.aiplatform.metadata import execution
+from google.cloud.aiplatform.metadata import metadata
+from google.cloud.aiplatform.metadata import metadata_store
+from google.cloud.aiplatform.metadata import resource
+from google.cloud.aiplatform.metadata import utils as metadata_utils
+from google.cloud.aiplatform.tensorboard import tensorboard_resource
+
+_LOGGER = base.Logger(__name__)
+
+
+@dataclass
+class _ExperimentRow:
+    """Class for representing a run row in an Experiments Dataframe.
+
+    Attributes:
+        params (Dict[str, Union[float, int, str]]): Optional. The parameters of this run.
+        metrics (Dict[str, Union[float, int, str]]): Optional. The metrics of this run.
+        time_series_metrics (Dict[str, float]): Optional. The latest time series metrics of this run.
+        experiment_run_type (Optional[str]): Optional. The type of this run.
+        name (str): Optional. The name of this run.
+        state (str): Optional. The state of this run.
+    """
+
+    params: Optional[Dict[str, Union[float, int, str]]] = None
+    metrics: Optional[Dict[str, Union[float, int, str]]] = None
+    time_series_metrics: Optional[Dict[str, float]] = None
+    experiment_run_type: Optional[str] = None
+    name: Optional[str] = None
+    state: Optional[str] = None
+
+    def to_dict(self) -> Dict[str, Union[float, int, str]]:
+        """Converts this experiment row into a dictionary.
+
+        Returns:
+            Row as a dictionary.
+        """
+        result = {
+            "run_type": self.experiment_run_type,
+            "run_name": self.name,
+            "state": self.state,
+        }
+        for prefix, field in [
+            (constants._PARAM_PREFIX, self.params),
+            (constants._METRIC_PREFIX, self.metrics),
+            (constants._TIME_SERIES_METRIC_PREFIX, self.time_series_metrics),
+        ]:
+            if field:
+                result.update(
+                    {f"{prefix}.{key}": value for key, value in field.items()}
+                )
+        return result
+
+
+class Experiment:
+    """Represents a Vertex AI Experiment resource."""
+
+    def __init__(
+        self,
+        experiment_name: str,
+        *,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        """
+
+        ```
+        my_experiment = aiplatform.Experiment('my-experiment')
+        ```
+
+        Args:
+            experiment_name (str): Required. The name of this experiment.
```

Comment:
Can this also be a full resource name, i.e. `projects/123/locations/us-central1/experiments/my-experiment`? If so, might be good to specify.